### ISSUE TITLE
[Question] What is the plan to workaround Google rate limiting

### ISSUE THREAD

**Issue by longnd** [myat-hsu-mon/web-scrapping-app#26]
Good decision was made to handle the scrapping process asynchronously by using a message queue to handle each keyword in the background separately. But the scrapping service didn't implement any mechanism to workaround Google Rate limiting, so the scraping process will likely be blocked. 
https://github.com/myat-hsu-mon/web-scrapping-app/blob/main/server/src/web-scrape/scrape.js
Do you have a plan to improve it and what it is?

**Comment by myat-hsu-mon** [comment]
Absolutely, thank you for sharing your feedback. In response to the issues you raised, I've implemented a delay mechanism before pushing messages to the queue for processing. I've also integrated the `rabbitmq-delayed-message-exchange` plugin. Currently, the system is sending two messages per second, which ensures that multiple messages won't be processed simultaneously. This approach effectively prevents excessive requests to Google. If you have any further suggestions or considerations, feel free to let me know!

**Comment by myat-hsu-mon** [comment]
Thank you for your valuable advice. 
I have previously experimented with rotating the user agents to simulate the requests by utilizing user-agents lib (for generating random user-agents) before. But, I encountered some limitations in its effectiveness. I think google has mechanisms to detect the user-agent being set, then the search results are not properly getting with the same keywords.
So, I chose using a delay mechanism as an alternative.

I really appreciate your insight.
